lib/rock/rest_api.rb

```diff
@@ -0,0 +1,142 @@
+require 'eventmachine'
+require 'rack'
+require 'thin'
+require 'orocos'
+require 'rack/static'
+require 'sprockets'
+require 'rock/webapp/tasks'
+
+module Rock
+    
+class RestApi
+        
+    def self.exec(argv = [])
```

Comment:
Don't pass ARGV to .exec. Leave the argv parsing in rock-rest-api, as e.g. the name service setup is *really* something that should NOT be done from within an API. Pass the relevant parameters as keyword arguments (e.g. host or cors_hosts).